apps/wolfsshd/configuration.c

```diff
@@ -474,85 +475,133 @@ static int HandleInclude(WOLFSSHD_CONFIG *conf, const char *value)
+                            break;
                         }
                     }
-                    if (postfix) {
-                        /* Skip if file is too short */
-                        if (WSTRLEN(dir->d_name) <= WSTRLEN(postfix)) {
-                            continue;
-                        }
-                        if (WSTRNCMP(dir->d_name + WSTRLEN(dir->d_name) -
-                                    WSTRLEN(postfix), postfix, WSTRLEN(postfix))
-                                == 0) {
-                            snprintf(filepath, PATH_MAX, "%s/%s", path,
-                                     dir->d_name);
-                        }
-                        else {
-                            /* Not a match */
-                            continue;
-                        }
+                    fileNames[j] = dir->d_name;
+                    i++;
+                }
+            }
+
+            for (i = 0; i < fileCount; i++) {
+                /* Check if filename prefix matches */
+                if (prefixLen > 0) {
+                    if ((int)WSTRLEN(fileNames[i]) <= prefixLen) {
+                        continue;
                     }
-                    else {
-                        snprintf(filepath, PATH_MAX, "%s/%s", path,
-                                 dir->d_name);
+                    if (WSTRNCMP(fileNames[i], prefix, prefixLen) != 0) {
+                        continue;
+                    }
+                }
+                if (postfix) {
+                    /* Skip if file is too short */
+                    if (WSTRLEN(fileNames[i]) <= WSTRLEN(postfix)) {
+                        continue;
                     }
-                    ret = wolfSSHD_ConfigLoad(conf, filepath);
-                    if (ret != WS_SUCCESS) {
-                        return ret;
+                    if (WSTRNCMP(fileNames[i] + WSTRLEN(fileNames[i]) -
+                                WSTRLEN(postfix), postfix, WSTRLEN(postfix))
+                            == 0) {
+                        WSNPRINTF(filepath, PATH_MAX, "%s/%s", path,
+                                 fileNames[i]);
+                    }
+                    else {
+                        /* Not a match */
+                        continue;
                     }
                 }
+                else {
+                    WSNPRINTF(filepath, PATH_MAX, "%s/%s", path,
+                             fileNames[i]);
+                }
+                ret = wolfSSHD_ConfigLoad(conf, filepath);
+                if (ret != WS_SUCCESS) {
+                    break;
+                }
             }
+            WFREE(fileNames, conf->heap, DYNTYPE_PATH);
+            WCLOSEDIR(&d);
         }
         else {
             /* Bad directory */
-            WFREE(filepath, NULL, 0);
-            return WS_BAD_ARGUMENT;
+            ret = WS_INVALID_PATH_E;
         }
-        WFREE(filepath, NULL, 0);
+        WFREE(path, conf->heap, DYNTYPE_PATH);
```

Comment:
Do  we need to check that path and filepath aren't NULL before doing these two `WFREE`s?